server/data_common/matrix_loader.py

```diff
@@ -19,32 +19,33 @@ def __init__(self, loader):
         self.data_adaptor = None
         self.data_lock = RWLock()
 
-    def acquire(self, app_config):
-        """returns the data_adaptor if cached.  opens the data_adaptor if not.
-        In either case, the a reader lock is taken.  Must call release when
-        the data_adaptor is no longer needed"""
-
+    def acquire_existing(self):
```

Comment:
this appears to be dead code?

Reply:
Kind of.  I  was thinking along the lines of your other comment about taking a read lock before leaving the critical section.  But then I didn't complete that part.  It's done now and fixes that issue.